io: in UpdateFromRule properly handle case of std::string

smuzaffar opened the pull request.

pcanal committed:
io: in UpdateFromRule properly handle case of std::string

cmsbuild commented:
A new Pull Request was created by  @smuzaffar for branch cms/master/8b3840f73b.

@cmsbuild, @iarspider, @smuzaffar can you please review it and eventually sign? Thanks.
@antoniovilela, @mandrenguyen, @rappoccio, @sextonkennedy you are the release manager for this.
cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"14d105abf3e488590b7455ab10a803167ed4e6ac":{"files":["core/meta/inc/TClassRef.h","core/meta/src/TClassRef.cxx"],"squashed":false,"time":1738680647},"a2647e1ecf2baa141fe541fbc41fe33d5af6a22f":{"files":["core/foundation/src/RConversionRuleParser.cxx"],"squashed":false,"time":1738735927},"d295ee0443a081eb0489dab764f351e3fe76a675":{"files":["io/io/src/TStreamerInfo.cxx"],"squashed":false,"time":1738634463}},"emoji":{},"last_seen_sha":"a2647e1ecf2baa141fe541fbc41fe33d5af6a22f","signatures":{}} -->

pcanal committed:
meta: TClassRef::fClassPtr needs to be atomic.

It can be read while being updated by another thread.

cmsbuild commented:
Pull request #216 was updated.

pcanal committed:
io rules: Normalize type name when recording sources

This fixes part (b) of the issue seen at:
   https://github.com/cms-sw/cmsdist/pull/9643#issuecomment-2633834328

The issue was due to the source type name not being known at run-time
because the type name was an alias which has not dictionary request
and thus was not known at the execution the header files have not been loaded
(and this is indeed the intent as we strive to run I/O without
loading headers)

cmsbuild commented:
Pull request #216 was updated.
